### PR TITLE
fix: auto-play of audio in more browsers

### PR DESCRIPTION
**Pull request by gustavkj**
Fixes: #4

Improves auto-play in more browsers by trying to play the audio multiple times after user the user interacted with the page.

### CODE REVIEW

File: src/js/script.js [hunk 1]
```diff
@@ -90,13 +111,40 @@ function audioSetup() {
...
+    // If playing the unmuted audio fails, mute it and play it.
+    audioFile.muted = true;
+
+    // Unmute it on any user interaction.
+    events.forEach((event) => {
+      document.addEventListener(event, playUnmuteAudio);
+    });
+
+    // This might still fail, maybe third time's the charm in `playUnmuteAudio`.
+    await audioFile.play();
```
Comment (lindell) [inline src/js/script.js:146]: We should avoid uncaught Exceptions?

File: src/js/script.js [hunk 2]
```diff
@@ -90,13 +111,40 @@ function audioSetup() {
...
+  function playUnmuteAudio() {
+    audioFile.muted = false;
+
+    // The muted play might have failed, then play it now.
+    if (audioFile.paused) {
+      /*
+        Note: For some reason Firefox still might fail here, but only on key
+        events if a special key is pressed (e.g. Control, Shift, etc).
+      */
+      audioFile.play();
```
Comment (lindell) [inline src/js/script.js:124]: We should avoid uncaught Exceptions?

Comment (lindell) [reply src/js/script.js:124]: We should not remove the listeners if firefox (or other browsers) fails?

Comment (gustavkj) [reply src/js/script.js:124]: Good point, probably not. 🙂

File: src/js/script.js [hunk 3]
```diff
@@ -90,12 +111,43 @@ function audioSetup() {
...
+  async function playUnmuteAudio() {
+    audioFile.muted = false;
+
+    // The muted play might have failed, then play it now.
+    if (audioFile.paused) {
+      /*
+        Note: For some reason Firefox still might fail here, but only on key
+        events if a special key is pressed (e.g. Control, Shift, etc).
+      */
+      await audioFile.play().catch(() => undefined);
```
Comment (lindell) [inline src/js/script.js:124]: Personal pref: `() => {}`

**Review by lindell — APPROVED** [review]

File: src/js/script.js (hunk 3 above)
Comment (gustavkj) [reply src/js/script.js]: Changed to that 👍

**Pull request merged**